cmd/view-fits/main.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2015 The astrogo Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"image"
+	"image/color"
+	"image/draw"
+	"log"
+	"os"
+
+	"github.com/astrogo/fitsio"
+
+	"golang.org/x/exp/shiny/driver"
+	"golang.org/x/exp/shiny/screen"
+	"golang.org/x/mobile/event/key"
+	"golang.org/x/mobile/event/paint"
+)
+
+var (
+	infos []fileInfo
+	help  = flag.Bool("help", false, "show help")
+)
+
+type fileInfo struct {
+	Name   string
+	Images []image.Image
+}
+
+func main() {
+
+	var r *os.File
+	var f *fitsio.File
+	var err error
+
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, `view-fits - a FITS image viewer.
+
+Usage of view-fits:
+$ view-fits [file-names]
+
+Examples:
+$ view-fits astrogo/fitsio/testdata/file-img2-bitpix+08.fits
+$ view-fits astrogo/fitsio/testdata/file-img2-bitpix+08.fits astrogo/fitsio/testdata/file-img2-bitpix+16.fits astrogo/fitsio/testdata/file-img2-bitpix+64.fits
+
+Navigation:
+right/left arrows: switch to previous/next file
+up/down arrows: switch to previous/next image in the current file
+`)
+	}
+
+	flag.Parse()
+
+	if *help || len(os.Args) < 2 {
+		flag.Usage()
+		os.Exit(0)
+	}
+
+	// Parsing input files.
+	for _, fname := range flag.Args() {
+
+		finfo := fileInfo{Name: fname}
+
+		// Opening the file.
+		if r, err = os.Open(fname); err != nil {
+			log.Fatal("Can not open the input file: %s.", err)
+		}
+
+		// Opening the FITS file.
+		if f, err = fitsio.Open(r); err != nil {
+			log.Fatal("Can not open the FITS input file: %s.", err)
+		}
+		defer f.Close()
+
+		// Getting the file HDUs.
+		hdus := f.HDUs()
+		for _, hdu := range hdus {
+			// Getting the header informations.
+			header := hdu.Header()
+			axes := header.Axes()
+
+			// Leaving HDU with no axes.
+			if len(axes) != 0 {
+				// Keeping HDU with IMAGE_HDU type.
+				if HDUType := hdu.Type(); HDUType == fitsio.IMAGE_HDU {
+					// Filling the images array.
+					finfo.Images = append(finfo.Images, hdu.(fitsio.Image).Image())
```

Comment:
the call to `fitsio.Image.Image()` may return `nil` if the image is invalid (`len(axes)!=2`, invalid `BITPIX` or invalid/non-sensical `width`/`height` values, etc...)
see:
 https://github.com/astrogo/fitsio/blob/master/image.go#L382

we should probably only append the returned `.Image()` if not-`nil`.